File: codes/yujo/121_besttimetobuyandsellstock.ts
function maxProfit(prices: number[]): number {
  let buyPrice = prices[0];
  let profit = 0;

  for (let i = 1; i < prices.length; i++) {
    const todayProfit = prices[i] - buyPrice;

    profit = Math.max(todayProfit, profit);
    buyPrice = Math.min(buyPrice, prices[i]);
  }

  return profit;
}

File: codes/hanjun/121_besttimetobuyandsellstock.py
def maxProfit(self, prices: List[int]) -> int:
    profit = [0]
    min_price = prices[0]
    
    for current_price in prices[1:]:
        if min_price > current_price:
            min_price = current_price
        else: profit.append(current_price - min_price)
        
    return max(profit)